pkg/objectupdate/sched/render.go

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ */
+
+package sched
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	SchedulerConfigFileName = "scheduler-config.yaml" // TODO duplicate from yaml
+	schedulerPluginName     = "NodeResourceTopologyMatch"
+)
+
+type ConfigCacheParams struct {
+	ResyncPeriodSeconds *int64
+}
+
+type ConfigParams struct {
+	Cache *ConfigCacheParams
+}
+
+func SchedulerConfig(cm *corev1.ConfigMap, schedulerName string, params *ConfigParams) error {
+	if cm.Data == nil {
+		return fmt.Errorf("no data found in ConfigMap: %s/%s", cm.Namespace, cm.Name)
+	}
+
+	data, ok := cm.Data[SchedulerConfigFileName]
+	if !ok {
+		return fmt.Errorf("no data key named: %s found in ConfigMap: %s/%s", SchedulerConfigFileName, cm.Namespace, cm.Name)
+	}
+
+	newData, err := RenderConfig([]byte(data), schedulerName, params)
+	if err != nil {
+		return err
+	}
+
+	cm.Data[SchedulerConfigFileName] = string(newData)
+	return nil
+}
+
+func RenderConfig(data []byte, schedulerName string, params *ConfigParams) ([]byte, error) {
```

Comment:
Is this change means that in case of updates to the API (mainly addition of new fields) we would need to update this function as well?